src/proto/device_path.rs

```diff
@@ -5,7 +5,7 @@ use crate::{proto::Protocol, unsafe_guid};
 /// Device path protocol.
 ///
 /// This can be opened on a `LoadedImage.device()` handle using the `HandleProtocol` boot service.
-#[repr(C)]
+#[repr(C, packed(1))]
```

Comment:
Agreed with what @josephlr said, would feel more consistent with just `packed`.

Reply:
Changed to use just `packed`. I expanded the docstring to note the alignment issue, and also to explain generally how Device Paths work since it initially confused the heck out of me :)